src/types.ts

```diff
@@ -291,11 +293,60 @@ export interface BlockExpression extends es.BaseExpression {
 
 export type substituterNodes = es.Node | BlockExpression
 
-export type TypeAnnotatedNode<T extends es.Node> = TypeAnnotation & T
+export {
+  Instruction as SVMInstruction,
+  Program as SVMProgram,
+  Address as SVMAddress,
+  Argument as SVMArgument,
+  Offset as SVMOffset,
+  SVMFunction
+} from './vm/svml-compiler'
+
+export type ContiguousArrayElementExpression = Exclude<es.ArrayExpression['elements'][0], null>
+
+export type ContiguousArrayElements = ContiguousArrayElementExpression[]
+
+// =======================================
+// Types used in type checker for type inference/type error checker for Source Typed variant
+// =======================================
+
+export enum PrimitiveType {
+  BOOLEAN = 'boolean',
+  NULL = 'null',
+  NUMBER = 'number',
+  STRING = 'string',
+  UNDEFINED = 'undefined'
+}
+
+export enum TSAllowedTypes {
+  ANY = 'any',
+  VOID = 'void'
+}
+
+export enum TSDisallowedTypes {
+  BIGINT = 'bigint',
+  NEVER = 'never',
+  OBJECT = 'object',
+  SYMBOL = 'symbol',
+  UNKNOWN = 'unknown'
+}
+
+export const TSBasicType = {
+  ...PrimitiveType,
+  ...TSAllowedTypes,
+  ...TSDisallowedTypes
+}
```

Comment:
```suggestion
export type PrimitiveType =
  | 'boolean'
  | 'null'
  | 'number'
  | 'string'
  | 'undefined';

export type TSAllowedTypes =
  | 'any'
  | 'void'

export type TSDisallowedTypes =
  | 'bigint'
  | 'never'
  | 'object'
  | 'symbol'
  | 'unknown'
```

Using union types might result in easier handling of types as compared to enums, and is consistent with the ESTree types.